underfs/s3/src/main/java/alluxio/underfs/s3/S3UnderFileSystem.java

```diff
@@ -492,15 +508,18 @@ private boolean isRoot(String key) {
       // managed "directories"
       S3Object[] objs = mClient.listObjects(mBucketName, path, "");
       if (recursive) {
-        String[] ret = new String[objs.length];
-        for (int i = 0; i < objs.length; i++) {
+        List<String> ret = new ArrayList<>();
```

Comment:
I think we should move to the `<>` operator since `Lists.newArrayList()` is mainly to make up for the lack of that shorthand.